Responder a comentarios elimindos, funcionando correctamente

YaelBadillo opened the pull request.

YaelBadillo committed:
Responder a comentarios elimindos, funcionando correctamente

AlexOrtegaZ commented on src/middlewares/is-sub-comment.middleware.js:
```diff
@@ -30,14 +29,15 @@ const isSubComment = async (req, res, next) => {
 
+  const manifestoCommentChildren = await manifestoCommentRepository.findAllByParentId(manifestoCommentParentId);
+  if (manifestoComment.deleted && manifestoCommentChildren.length === 0) {
+    return next(new ApiError(httpStatus.NOT_FOUND, 'The parent comment does not have any sub comments'));
```
El http status debería ser BAD_REQUEST no NOT_FOUND
El nombre de esta función dice que checa si el comentario es un sub comentario, pero aqui no no solo checa eso si no que tambien checa si el comentario padre esta eliminado y si tiene mas subcomentarios.
En este caso se tienen que hacer mas de una función para validar cada uno de estos casos con nombres distintos.

YaelBadillo replied:
Se le cambió el nombre al middleware como al resto de los demás.

YaelBadillo committed:
Refact: Rename is-subcomment to can-create-subcomment like the rest.

Furthermore, change http status to bad-requrest and change message error to another more specific.
